src/main/java/de/tum/in/www1/artemis/domain/plagiarism/PlagiarismVerdict.java

```diff
@@ -1,6 +1,24 @@
 package de.tum.in.www1.artemis.domain.plagiarism;
 
+import java.util.Collection;
+import java.util.Comparator;
+import java.util.Objects;
+
 public enum PlagiarismVerdict {
 
-    PLAGIARISM, POINT_DEDUCTION, WARNING, NO_PLAGIARISM
+    // Warning: The ordering of the enum values are important, they should be ordered from the most severe to least severe.
```

Comment:
Maybe you should give them numerical values (200, 150, 100, 50). That way the ordering stays consistent and new verdicts can easily be inserted in between later on.

Reply:
In order to add some numerical values, we need to add a new field to the class (unlike Typescript where we can write `PLAGIARISM = 200` etc.) which would complicate it further than it should in my opinion. Besides, since both database and json serializer uses string representation of the enum values, adding new verdicts in between is easy.

Only important point is that if someone is adding a new enum value, they should add it to the correct place.